.github/workflows/validate.yaml

```diff
@@ -39,6 +39,22 @@ jobs:
         with:
           path: ./src/github.com/${{ github.repository }}
 
+      - name: Merge upstream
+        if: github.event_name == 'pull_request'
+        with:
+          path: ./src/github.com/${{ github.repository }}
+        run: |
+          if ! git config user.name > /dev/null; then
+            git config user.name "John Doe"
+          fi
+          if ! git config user.email > /dev/null; then
+            git config user.email "johndoe@localhost"
+          fi
+          git remote add upstream https://github.com/${{ github.repository }}.git
+          git fetch upstream ${{ github.base_ref }}
+          git pull --no-rebase --no-commit upstream ${{ github.base_ref }}
```

Comment:
This is equivalent to git merge right?

Reply:
Yes

Reply:
Where do we do the push?

Reply:
We don't, this is basically faking the fact that validates run on target branch + PR changes because without it a GH action will run with the content of your branch which might not be based on the latest commit.

Reply:
Correct ok it is just the validation ;)